docs/hyperopt.md

```diff
@@ -314,6 +322,87 @@ There are four parameter types each suited for different purposes.
 !!! Warning
     Hyperoptable parameters cannot be used in `populate_indicators` - as hyperopt does not recalculate indicators for each epoch, so the starting value would be used in this case.
 
+### Optimizing an indicator parameter
+
+Assuming you have a simple strategy in mind - a EMA cross strategy (2 Moving averages crossing) - and you'd like to find the ideal parameters for this strategy.
+
+``` python
+import talib.abstract as ta
+
+from freqtrade.strategy import IStrategy
+from freqtrade.strategy import CategoricalParameter, DecimalParameter, IntParameter
+import freqtrade.vendor.qtpylib.indicators as qtpylib
+
+class MyAwesomeStrategy(IStrategy):
+    stoploss = 0.5
+    timeframe = '15m'
+    # Define the parameter spaces
+    buy_ema_short = IntParameter(3, 50, default=5)
+    buy_ema_long = IntParameter(15, 200, default=50)
```

Comment:
This made me think of a common missing feature I come across:
Range is 15-200, but not every single value in between this range is valuable. So a `step` or `distance` parameter like with `FloatParameter` would be extremely useful...

Reply:
You can do stepping:
```python
step = 2
buy_ema_short = IntParameter(2 / step, 50 / step, default=5)
def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
    for val in self.buy_ema_short.range:
        dataframe[f'ema_short_{val}'] = ta.EMA(dataframe, timeperiod=val*step)
```

Reply:
while it works, you must make sure to use it in all cases ... so i'd consider this as "dangerous" for now :laughing:

Reply:
^True, end if you do use the workaround then the hyperopt results might need manual multiplication.
Cool feature if it existed.